event.go

```diff
@@ -230,3 +248,89 @@ func (e Event) KeyIDs(signingName string) []string {
 func (e Event) Verify(signingName, keyID string, publicKey ed25519.PublicKey) error {
 	return verifyEventSignature(signingName, keyID, publicKey, e.eventJSON)
 }
+
+// IsState returns true if the event is a state event.
+func (e Event) IsState() bool {
+	return e.fields.StateKey != nil
+}
+
+// StateKey returns the "state_key" of the event, or the empty string if it's
+func (e Event) StateKey() string {
```

Comment:
Comment isn't finished. Also, it feels deeply wrong to be defaulting to the empty string when the empty string is used so much as a state key. I feel this buys us nothing, since you need to know if this event is a state event in order for this to make any sense, so the typical use will be:
```go
if e.IsState() {
    key := e.StateKey()
}
```

...at which point you might as well be doing the `nil` deference yourself. What exactly is the intended use case here?